lib/internal/Magento/Framework/Css/PreProcessor/Adapter/CssInliner.php

```diff
@@ -6,67 +6,37 @@
 namespace Magento\Framework\Css\PreProcessor\Adapter;
 
 use Magento\Framework\App\State;
-use Pelago\Emogrifier;
+use Pelago\Emogrifier\CssInliner as EmogrifierCssInliner;
+use Symfony\Component\CssSelector\Exception\ParseException;
 
 /**
  * This class will inline the css of an html to each tag to be used for applications such as a styled email.
  */
 class CssInliner
 {
     /**
-     * @var Emogrifier
+     * @var State
      */
-    private $emogrifier;
+    private $appState;
 
     /**
      * @param State $appState
      */
     public function __construct(State $appState)
     {
-        $this->emogrifier = new Emogrifier();
-        $this->emogrifier->setDebug($appState->getMode() === State::MODE_DEVELOPER);
+        $this->appState = $appState;
     }
 
     /**
-     * Sets the HTML to be used with the css. This method should be used with setCss.
-     *
      * @param string $html
-     * @return void
-     */
-    public function setHtml($html)
```

Comment:
It would be better to preserve the interfaces of framework classes.

Let's keep the existing interface. This can be done by saving the data that is passed to setters to corresponding properties and then using these properties when the `process` method is called